test/common/quic/client_connection_factory_impl_test.cc

```diff
@@ -107,7 +112,15 @@ TEST_P(QuicNetworkConnectionTest, Srtt) {
       dispatcher_, test_address_, test_address_, quic_stat_names_, rtt_cache, store_);
 
   EnvoyQuicClientSession* session = static_cast<EnvoyQuicClientSession*>(client_connection.get());
-
+  EXPECT_CALL(os_sys_calls, supportsGetifaddrs())
```

Comment:
hi Dan, thanks for your comments, I have same suspicious, I have tried as below:
I comment out rest of the code in this testcase, only leave the EXPECT_CALL, it will pass anyway. 
If I replace the EXPECT_CALL.Times with EXPECT_CALL.WillOnce, it will fail, no matter I put it anywhere. 
With the tests, I think the syscall will not be invoked unless I mock a config "allow_syscall_for_interface_name_", I am trying to find the way to mock it, and do you happen know some example code to mock the configurations?
Thank you.

Reply:
hope this is what you looking for https://github.com/envoyproxy/envoy/blob/e4ee794cf4e39d3dfac9179e03c0d61f1d2f7b1c/test/extensions/filters/listener/tls_inspector/tls_inspector_test.cc#L66-L67

Reply:
thanks Alex, I did miss this line, and I found I need add some code to it, will update later.

Reply:
@danzh2010 @RyanTheOptimist 
Hi,
TO fulfill the functionality of #20209, I think we need to expend the createQuicNetworkConnection() to retrieve a knob from cluster's configuration "set_local_interface_name_on_upstream_connections", I think the direct way is to add a parameter to the function, but I see there is another TODO set by Dan "move this temporary config and initial RTT configuration to h3 pool", is there any ongoing plan to change the parameters, if yes, I'd better sync with the design.

Thank you

Reply:
You are right that set_local_interface_name_on_upstream_connections config knob should be passed into createQuicNetworkConnection(). And its caller, h3 pool, already has access to clusterInfo.

The TODO is for a different matter, and no on-going plan to change the function interface.

Reply:
> You are right that set_local_interface_name_on_upstream_connections config knob should be passed into createQuicNetworkConnection(). And its caller, h3 pool, already has access to clusterInfo.
> 
> The TODO is for a different matter, and no on-going plan to change the function interface.

thank you Dan, I will make some changes to the code